### PR TITLE
Adding extra balance feature

### PR DESCRIPTION
**Pull request by AbdoHamed-TheBadNomads**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -79,4 +79,15 @@ async def IsMemberWorking(ctx, discorduser, date = datetime.today().strftime('%d
 
     await ctx.send(content = "Done", delete_after = 0.1)
 
+@slash.slash(name = "InsertExtraLeaveBalance", description = "Inserts an extra credit for the provided leave type", options = UI.CreateInsertExtraBalanceOptions(), guild_ids = guild_ids)
```
Comment (AntWong-TheBadNomads) [inline main.py:82]: Let's just call this creditLeaves

File: UI.py [hunk 2]
```diff
@@ -208,6 +208,37 @@ def CreateIsMemberWorkingOptions():
...
+            description = "leave type",
+            option_type = SlashCommandOptionType.STRING,
+            required = True,
+            choices = CreateLeaveTypeChoices()
+        ),
+        create_option(
+            name = "dayscount",
+            description = "amount of extra credit (can be negative number)",
+            option_type = SlashCommandOptionType.INTEGER,
+            required = True
```
Comment (AntWong-TheBadNomads) [inline UI.py:230]: not required, default to 1

File: main.py (hunk 1 above)
Comment (AbdoHamed-TheBadNomads) [reply main.py:82]: Done

File: UI.py (hunk 2 above)
Comment (AbdoHamed-TheBadNomads) [reply UI.py]: Done

File: UI.py [hunk 3]
```diff
@@ -208,6 +208,37 @@ def CreateIsMemberWorkingOptions():
...
+            name = "leavetype",
+            description = "leave type",
+            option_type = SlashCommandOptionType.STRING,
+            required = True,
+            choices = CreateLeaveTypeChoices()
+        ),
+        create_option(
+            name = "dayscount",
+            description = "amount of extra credit (can be negative number)",
+            option_type = SlashCommandOptionType.INTEGER,
```
Comment (AntWong-TheBadNomads) [inline UI.py:229]: can we input floats if this is integer?

Comment (AbdoHamed-TheBadNomads) [reply UI.py]: Done

File: UI.py [hunk 4]
```diff
@@ -213,6 +213,38 @@ async def UpdateLeaveEmbed(member, message, embed, newStatus):
...
+        create_option(
+            name = "leavetype",
+            description = "leave type",
+            option_type = SlashCommandOptionType.STRING,
+            required = True,
+            choices = CreateLeaveTypeChoices()
+        ),
+        create_option(
+            name = "dayscount",
+            description = "amount of extra credit (can be negative number)",
```
Comment (AntWong-TheBadNomads) [inline UI.py:234]: Probably important to note that it defaults to 1 here

Comment (AbdoHamed-TheBadNomads) [reply UI.py]: Done

File: UI.py [hunk 5]
```diff
@@ -213,6 +213,38 @@ async def UpdateLeaveEmbed(member, message, embed, newStatus):
...
+        create_option(
+            name = "leavetype",
+            description = "leave type",
+            option_type = SlashCommandOptionType.STRING,
+            required = True,
+            choices = CreateLeaveTypeChoices()
+        ),
+        create_option(
+            name = "dayscount",
+            description = "amount of extra credit (can be negative number) leave empty for 1 day",
```
Comment (AntWong-TheBadNomads) [inline UI.py:234]:
```suggestion
            description = "amount of extra leaves to credit (can be a negative number). Defaults to 1",
```

Comment (AbdoHamed-TheBadNomads) [reply UI.py]: Done

**Pull request merged**